modules/apps/message-boards/message-boards-service/src/main/java/com/liferay/message/boards/service/impl/MBMessageLocalServiceImpl.java

```diff
@@ -1470,6 +1474,124 @@ public int getRootDiscussionMessagesCount(
 		return count;
 	}
 
+	@Override
+	public List<MBMessage> getSiteUserMessageBoardMessagesActivity(
+		long groupId, long userId, int start, int end) {
+
+		MBMessageTable aliasMBMessageTable1 = MBMessageTable.INSTANCE.as(
+			"aliasMBMessageTable1");
+
+		DSLQuery dslQuery = DSLQueryFactoryUtil.select(
+			MBMessageTable.INSTANCE
+		).from(
+			MBMessageTable.INSTANCE
+		).where(
+			MBMessageTable.INSTANCE.modifiedDate.in(
+				DSLQueryFactoryUtil.select(
+					DSLFunctionFactoryUtil.max(
+						MBMessageTable.INSTANCE.modifiedDate)
+				).from(
+					aliasMBMessageTable1
+				).where(
+					MBMessageTable.INSTANCE.rootMessageId.eq(
+						MBMessageTable.INSTANCE.parentMessageId
+					).and(
+						MBMessageTable.INSTANCE.groupId.eq(
+							groupId
+						).and(
+							MBMessageTable.INSTANCE.userId.eq(userId)
+						)
+					).and(
+						MBMessageTable.INSTANCE.categoryId.eq(
+							aliasMBMessageTable1.categoryId)
+					)
+				)
+			).or(
+				MBMessageTable.INSTANCE.parentMessageId.eq(
+					0L
+				).and(
+					MBMessageTable.INSTANCE.rootMessageId.notIn(
```

Comment:
Try not use notIn.
Is there not exists clause?
If there is, I think better than notIn because of performance.

Reply:
I looked for it but it doesn't have "doesn't exist" @joseabelenda